### PR TITLE
feat: Update `artifacthub-pkg.yml` automatically from now

### PR DESCRIPTION
**Pull request by viccuad**
## Description

Relates to https://github.com/kubewarden/kubewarden-controller/issues/325.

## Test

See CI run on this RP, and when the 0.1.8 tag is created.

### Potential improvement

- [ ] TODO tag this as 0.1.8 in main, once it is merged.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -11,7 +11,7 @@ jobs:
 
   test:
     name: run tests and linters
-    uses: kubewarden/github-actions/.github/workflows/reusable-test-policy-go.yml@v1
+    uses: kubewarden/github-actions/.github/workflows/reusable-test-policy-go.yml@v2
```
Comment (jvanz) [inline .github/workflows/release.yml:14]:
```suggestion
    uses: kubewarden/github-actions/.github/workflows/reusable-test-policy-go.yml@v2.0.2
```

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -23,6 +23,6 @@ jobs:
       # Required by cosign keyless signing
       id-token: write
 
-    uses: kubewarden/github-actions/.github/workflows/reusable-release-policy-go.yml@v1
+    uses: kubewarden/github-actions/.github/workflows/reusable-release-policy-go.yml@v2
```
Comment (jvanz) [inline .github/workflows/release.yml:26]:
```suggestion
    uses: kubewarden/github-actions/.github/workflows/reusable-release-policy-go.yml@v2.0.2
```

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -3,4 +3,4 @@ name: Continuous integration
 jobs:
   test:
     name: run tests and linters
-    uses: kubewarden/github-actions/.github/workflows/reusable-test-policy-go.yml@v1
+    uses: kubewarden/github-actions/.github/workflows/reusable-test-policy-go.yml@v2
```
Comment (jvanz) [inline .github/workflows/test.yml:6]:
```suggestion
    uses: kubewarden/github-actions/.github/workflows/reusable-test-policy-go.yml@v2.0.2
```

File: Makefile [hunk 4]
```diff
@@ -1,11 +1,23 @@
 SOURCE_FILES := $(shell find . -type f -name '*.go')
+VERSION := $(shell git describe --tags --abbrev=0 | cut -c2-)
```
Comment (jvanz) [inline Makefile:2]:
```suggestion
VERSION := $(shell git describe | cut -c2-)
```

File: Makefile [hunk 5]
```diff
@@ -1,11 +1,23 @@
...
+		tinygo build -o policy.wasm -target=wasi -no-debug .
 
-annotated-policy.wasm: policy.wasm metadata.yml
-	kwctl annotate -m metadata.yml -o annotated-policy.wasm policy.wasm
+artifacthub-pkg.yml: metadata.yml go.mod
+	kwctl scaffold artifacthub \
+	    --metadata-path metadata.yml --version $(VERSION) \
+		--questions-path questions-ui.yml > artifacthub-pkg.yml.tmp \
+	&& mv artifacthub-pkg.yml.tmp artifacthub-pkg.yml \
+	|| rm -f artifacthub-pkg.yml.tmp
```
Comment (jvanz) [inline Makefile:17]:
```suggestion
        $(warning If you are updating the artifacthub-pkg.yml file for a release, \
                remember to set the VERSION variable with the proper value. \
                To use the latest tag, use the following command:  \
                make VERSION=$$(git describe --tags --abbrev=0 | cut -c2-) annotated-policy.wasm)
        kwctl scaffold artifacthub \
            --metadata-path metadata.yml --version $(VERSION) \
                --questions-path questions-ui.yml --output artifacthub-pkg.yml 
```

**Review by jvanz — CHANGES_REQUESTED** [review]
During the last planning we proposed some changes in the PRs updating the artifacthub files. I've opened a bunch of comments with these changes.

File: Makefile (hunk 5 above)
Comment (viccuad) [reply Makefile]: done in https://github.com/kubewarden/hostpaths-psp-policy/pull/14/commits/1fe372d56cf24a2234f517c565170aee49363ae9.

File: Makefile (hunk 4 above)
Comment (viccuad) [reply Makefile]: done in https://github.com/kubewarden/hostpaths-psp-policy/pull/14/commits/1fe372d56cf24a2234f517c565170aee49363ae9.

File: .github/workflows/release.yml (hunk 2 above)
Comment (viccuad) [reply .github/workflows/release.yml]: done in https://github.com/kubewarden/hostpaths-psp-policy/pull/14/commits/1fe372d56cf24a2234f517c565170aee49363ae9.

File: Makefile [hunk 6]
```diff
@@ -1,11 +1,28 @@
...
+		--rm \
+		-e GOFLAGS="-buildvcs=false" \
+		-v ${PWD}:/src \
+		-w /src tinygo/tinygo:0.23.0 \
+		tinygo build -o policy.wasm -target=wasi -no-debug .
 
-annotated-policy.wasm: policy.wasm metadata.yml
-	kwctl annotate -m metadata.yml -o annotated-policy.wasm policy.wasm
+
+artifacthub-pkg.yml: metadata.yml go.mod
```
Comment (viccuad) [inline Makefile:15]: we are missing adding `artifacthub-pkg.yml` file to the `make clean` target.

File: artifacthub-pkg.yml [hunk 7]
```diff
@@ -1,43 +1,42 @@
----
-version: 0.1.7
+# Kubewarden Artifacthub Package config
+#
+# Use this config to submit the policy to https://artifacthub.io.
+#
+# This config can be saved to its default location with:
+#   kwctl scaffold artifacthub > artifacthub-pkg.yml 
+version: 0.1.7-9-g1fe372d
```
Comment (jvanz) [inline artifacthub-pkg.yml:7]: I think it is necessary to regenerate the file after changing the command to get the value for the `VERSION`  variable

**Review by jvanz — CHANGES_REQUESTED** [review]

**Pull request merged**